packages/data-service/src/connect-mongo-client.spec.ts

```diff
@@ -135,9 +137,15 @@ describe('connectMongoClient', function () {
           /(All configured authentication methods failed|ENOTFOUND compass-tests\.fakehost\.localhost)/
         );
 
-        expect(
-          (await tryConnect(tunnelLocalPort).catch((err) => err)).code
-        ).to.equal('ECONNREFUSED');
+        for (let i = 0; i < 10; i++) {
```

Comment:
can this get flaky?

Reply:
I don’t think so – I’ve written similar things for the same kind of scenario in Node.js core, and have not seen issues with that.

This isn’t quite like a timeout, because the steps required to get the “server closed” notification do happen in fixed event loop iterations. The only reason I’m specifying `10` here as a number is that Node.js does not make strict guarantees about how many iterations that is (e.g. it could introduce another internal event loop iteration for closing network sockets without considering that a breaking change); so it’s more of a “number that is sufficiently big to never need to be changed”.

Reply:
ahh ok makes sense!